hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -603,7 +652,23 @@ extends Importer:
     case id: Ident =>
       Param(FldFlags.empty, fieldOrVarSym(ParamBind, id), N) :: Nil
     case InfixApp(lhs: Ident, Keyword.`:`, rhs) =>
-      Param(FldFlags.empty, fieldOrVarSym(ParamBind, lhs), S(term(rhs))) :: Nil
+      // return S(moduleParam) if t represents a module parameter
+      // return N otherwise
+      def moduleParam(t: Term): Opt[Param] = t match
+        case s: hkmc2.semantics.Term.Sel => s.symbol
+          .flatMap(_.asMod)
+          .map(_ => Param(FldFlags.module, fieldOrVarSym(ParamBind, lhs), S(s)))
+        case hkmc2.semantics.Term.TyApp(s: hkmc2.semantics.Term.Sel, _) => s.symbol
+          .flatMap(_.asMod)
+          .map(_ => Param(FldFlags.module, fieldOrVarSym(ParamBind, lhs), S(s)))
+        case _ => N
```

Comment:
Why not just use `s.symbol` directly?

Reply:
IIRC `s.symbol` is only defined on either `s: Term.Sel` or `s: Term.Ref`. I had to check if the type really is a module, so I call `asMod` and see if I can get some module, if so, I create a `Param` with module flag being set; otherwise, I return `N` and fallback to a regular `Param`.